File: pset2/caesar.c
// Shift cipher
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

char change(char wordstr,int b);

int main(int argc, string argv[])
{
    // For no inputs
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    int a = atoi(argv[1]) % 26;
    // Userinput for non-numbers and wrong key inputs
    for (int i = 0;i < strlen(argv[1]);i++)
    {
        if (isdigit(argv[1][i]) == 0)
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }
    // Body
    string wordstr = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0;i < strlen(wordstr);i++)
    {
        if (islower(wordstr[i]))
        {
            wordstr[i] = change(wordstr[i], a);
        }
        else if (isupper(wordstr[i]))
        {
            wordstr[i] = change(wordstr[i], a);
        }
        printf("%c", wordstr[i]);
    }
    printf("\n");
}

char change(char wordstr, int b)
{
    // Lowercase
    if (wordstr + b > 'Z' && wordstr + b > 'z')
    {
        wordstr += b - 26;
    }
    // Uppercase
    else if (wordstr + b > 'Z' && wordstr + b < 'a')
    {
        wordstr += b - 26;
    }
    else
    {
        wordstr += b; 
    }
    return wordstr;
}

File: pset6/caesar (python)/caesar.py
from cs50 import get_string
from sys import argv

if len(argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)
    
key = int(argv[1],10) % 26
input = get_string("plaintext: ")
print("ciphertext: ", end="")

for char in input:
    if char.isalpha():
        new = ord(char) + key
        if (new > 90 and new < 97) or (new > 90 and new > 122):
            new = new - 26
        newer = chr(new)
    else:
        newer = char

    
    print(f"{newer}", end="")
print("")